### ISSUE TITLE
None synch frame download.

### ISSUE THREAD

**Issue by sasmaster** [diederickh/video_streamer#1]
Hi!Great lib you have here.I am looking ath YUV grabber and wonder if I need hasNewFrame() if I don't use vsync? I mean I rendere frames  in OpenGL as fast as possible and once I rendered the next frame I want to download it immediately.I don't exactly understand the idea behind hasNewFrame().Is it to keep fixed  frame rate delay intact>? Thanks.

**Comment by roxlu** [comment]
Hi sasmaster,  that's exactly where hasNewFrame() is used for. To make sure audio/video is correctly in sync
you want to provides video data at a steady pace. I haven't got time to cleanup this library after I created it for a project. It's lacking some documentation.

**Comment by sasmaster** [comment]
Yeah , got it.Removed hasNewFrame() and it writes frames all right without
vsynch .Great,thanks.Btw,have you considered YUV frame composition part
doing on CUDA or Compute shader?

On Mon, Jan 27, 2014 at 11:58 AM, roxlu notifications@github.com wrote:

> Hi sasmaster, that's exactly where hasNewFrame() is used for. To make sure
> audio/video is correctly in sync
> you want to provides video data at a steady pace. I haven't got time to
> cleanup this library after I created it for a project. It's lacking some
> documentation.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/roxlu/video_streamer/issues/1#issuecomment-33354056
> .

## 

Michael Ivanov
Independent Pixel Commander
onlygraphix.com
Tel:+972 54 4962254

**Comment by roxlu** [comment]
Hi Michael, lets continue this talk on skype / gmail

On Mon, Jan 27, 2014 at 11:28 AM, Michael Ivanov
notifications@github.comwrote:

> Yeah , got it.Removed hasNewFrame() and it writes frames all right without
> vsynch .Great,thanks.Btw,have you considered YUV frame composition part
> doing on CUDA or Compute shader?
> 
> On Mon, Jan 27, 2014 at 11:58 AM, roxlu notifications@github.com wrote:
> 
> > Hi sasmaster, that's exactly where hasNewFrame() is used for. To make
> > sure
> > audio/video is correctly in sync
> > you want to provides video data at a steady pace. I haven't got time to
> > cleanup this library after I created it for a project. It's lacking some
> > documentation.
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/roxlu/video_streamer/issues/1#issuecomment-33354056>
> > .
> 
> ## 
> 
> Michael Ivanov
> Independent Pixel Commander
> onlygraphix.com
> Tel:+972 54 4962254
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/roxlu/video_streamer/issues/1#issuecomment-33356188
> .

## 

Apollo +++++++++
Interactive Media
+++++++++++++++
Diederick Huijbers ===

# diederick@apollomedia.nl

Zeeburgerpad 74 ::::::::
1019 AD Amsterdam
mobile 06 - 12 44 09 22
phone 020 - 707 78 96
//\//\//\//\//\//\//\//\//\
www.apollomedia.nl +++
++++++++++++++++
